src/Standard.Licensing.Tests/KeyGeneratorTests.cs

```diff
@@ -47,10 +46,10 @@ public void Ensure_To_Not_Generate_Identical_Keys()
 
             for (int i = 0; i < 100; i++)
             {
-                var keyGenerator = new KeyGenerator(256); //default key size
-                var pair = keyGenerator.GenerateKeyPair();
-                var privateKey = pair.ToEncryptedPrivateKeyString(passPhrase);
-                var publicKey = pair.ToPublicKeyString();
+                KeyGenerator keyGenerator = new KeyGenerator(256); //default key size
+                KeyPair? pair = keyGenerator.GenerateKeyPair();
+                string? privateKey = pair.ToEncryptedPrivateKeyString(passPhrase);
+                string? publicKey = pair.ToPublicKeyString();
```

Comment:
Same thing here

Reply:
In C#, the ? after a type denotes a nullable value type. This means that the variable can hold either a value of that type or null. This is particularly useful for value types (like int, double, bool, etc.) which cannot naturally be null.
Here are some reasons why you might want to use a nullable type:

1. **Representing Absence of Value:** It allows you to represent the absence of a value in a way that's consistent with how reference types work (where null can be used). For example, int? can represent an integer that might not have a value.
2. **Default Values:** For value types, the default value is generally 0, false, or another default. A nullable type allows you to explicitly indicate that the value is not set or unknown by setting it to null.
3. **Database Interactions:** When working with databases, many fields can have null values. Using nullable types ensures your C# code can accurately represent and handle these scenarios.
4. **Conditional Logic:** You can easily check if a nullable type has a value using .HasValue and retrieve the value using .Value. 
5. **Enhanced Readability and Maintenance:** By explicitly declaring a type as nullable, it makes the code more readable and maintainable, clearly indicating that the variable is allowed to have no value.

Reply:
@ammarheidari yes, this is the syntax for a nullable type, but the along with the change to coding style here, you changed the nullability of the values. For instance `GenerateKeyPair()` does not return a nullable value, so along with teh style change, this is a behavioral change.